lib/commands/command-reshim.bash

```diff
@@ -103,9 +103,12 @@ generate_shim_for_executable() {
 generate_shims_for_version() {
   local plugin_name=$1
   local full_version=$2
-  for executable_path in $(plugin_executables "$plugin_name" "$full_version"); do
+  local executable_path
+  local all_plugin_executables
+  all_plugin_executables="$(plugin_executables "$plugin_name" "$full_version")"
```

Comment:
Aren't these executable paths delimited by newlines?

Couldn't we just use the existing code with `IFS=$'\n'` set?

Reply:
We can't, because `for` is a reserved word and not a real command; I've changed this to use the form taken [elsewhere in `utils.bash`](https://github.com/asdf-vm/asdf/blob/60e0e05a9f289e10514890a905c05f8573eb2b7c/lib/utils.bash#L692) instead, though.

Reply:
@mook-as to clarify, is `for` a reserved word in POSIX shells?

Reply:
I didn't know this and had to look it up, but yes, `for` is a [reserved word](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_04) for POSIX shells.  See also the [definition of a for loop](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_09_04_030).

I did know that it wasn't working with `FOO=bar for …`, at least :)